tests/bgp/test_ipv6_nlri_over_ipv4.py

```diff
@@ -0,0 +1,194 @@
+    time.sleep(30)
+
+    # clear BGP table
+    cmd = 'vtysh -n {} -c "clear ip bgp * soft"'.format(setup['namespace'])
+    setup['duthost'].shell(cmd)
+    cmd = 'vtysh -c "clear ip bgp * soft"'
+    setup['neighhost'].shell(cmd)
+
+    # verify route is no longer shared
+    cmd = "show ipv6 route {}".format(setup['dut_nlri_route'])
+    dut_route_out = setup['duthost'].shell(cmd)['stdout']
+    assert setup['neigh_ip_v6'] not in dut_route_out
+    cmd = "show ipv6 route {}".format(setup['neigh_nlri_route'])
+    neigh_route_out = setup['neighhost'].shell(cmd)['stdout']
+    assert setup['dut_ip_v6'] not in neigh_route_out
+
+    # configure IPv4 peer config on DUT
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "neighbor NLRI peer-group" -c "address-family ipv4 unicast" \
+        -c "neighbor NLRI allowas-in" -c "neighbor NLRI send-community both" \
+        -c "neighbor NLRI soft-reconfiguration inbound" -c "exit-address-family" -c "address-family ipv6 unicast" \
+        -c "neighbor NLRI allowas-in" -c "neighbor NLRI send-community both" \
+            -c "neighbor NLRI soft-reconfiguration inbound"'.format(setup['namespace'], setup['dut_asn'])
+    setup['duthost'].shell(cmd, module_ignore_errors=True)
+
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "neighbor {} peer-group NLRI" -c "neighbor {} remote-as {}"\
+        -c "address-family ipv4 unicast" -c "neighbor NLRI activate" -c "exit-address-family" \
+        -c "address-family ipv6 unicast" -c "neighbor NLRI activate"'\
+            .format(setup['namespace'], setup['dut_asn'], setup['neigh_ip_v4'], setup['neigh_ip_v4'],
+                    setup['neigh_asn'])
+    setup['duthost'].shell(cmd, module_ignore_errors=True)
+    logger.info("DUT BGP Config After Peer Config: {}".format(setup['duthost'].shell('show run bgp')['stdout']))
+
+    # configure IPv4 peer on neighbor
+    cmd = 'vtysh -c "config" -c "router bgp {}" -c "neighbor NLRI peer-group" -c "address-family ipv4 unicast" \
+        -c "neighbor NLRI allowas-in" -c "neighbor NLRI send-community both" \
+            -c "neighbor NLRI soft-reconfiguration inbound" -c "address-family ipv6 unicast" \
+                -c "neighbor NLRI activate" -c "neighbor NLRI allowas-in" \
+         -c "neighbor NLRI soft-reconfiguration inbound"'\
+            .format(setup['neigh_asn'])
+    setup['neighhost'].shell(cmd, module_ignore_errors=True)
+
+    cmd = 'vtysh -c "config" -c "router bgp {}" -c "neighbor {} peer-group NLRI" -c "neighbor {} remote-as {}"\
+        -c "address-family ipv4 unicast" -c "neighbor NLRI activate"'.format(setup['neigh_asn'], setup['dut_ip_v4'],
+                                                                             setup['dut_ip_v4'], setup['dut_asn'])
+    setup['neighhost'].shell(cmd, module_ignore_errors=True)
+    logger.info("Neighbor BGP Config After Peer Config: {}".format(setup['neighhost'].shell('show run bgp')['stdout']))
+
+    time.sleep(30)
```

Comment:
Same as above, better to wait and check to see if the neighbor is established.

Reply:
@lolyu I have made this change.